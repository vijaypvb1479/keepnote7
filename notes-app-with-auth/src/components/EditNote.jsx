import React, { Component } from 'react';
import Button from '@material-ui/core/Button';
import TextField from '@material-ui/core/TextField';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogTitle from '@material-ui/core/DialogTitle';
import { withStyles } from '@material-ui/core';
import ColorPicker from 'react-color';
import ColorLensIcon from '@material-ui/icons/ColorLens';

const styles = theme => ({
    swatch: {
        padding: '5px',
        background: '#fff',
        borderRadius: '1px',
        boxShadow: '0 0 0 1px rgba(0,0,0,.1)',
        display: 'inline-block',
        cursor: 'pointer',
    },
    popover: {
        position: 'fixed',
        zIndex: '2',
    },
});

class EditNote extends Component {
    constructor(props) {
        super(props);
        this.state = {
            open: true,
            noteTitle: this.props.notes.filter(note => note.id === parseFloat(this.props.match.params.id))[0].noteTitle,
            noteDescription: this.props.notes.filter(note => note.id === parseFloat(this.props.match.params.id))[0].noteDescription,
            color: this.props.notes.filter(note => note.id === parseFloat(this.props.match.params.id))[0].color,
            background: 'black',
        }
        this.handleClickOpen = this.handleClickOpen.bind(this);
        this.handleUpdateNote = this.handleUpdateNote.bind(this);
        this.handleClose = this.handleClose.bind(this);
        this.handleNoteTitleChange = this.handleNoteTitleChange.bind(this);
        this.handleNoteDescriptionChange = this.handleNoteDescriptionChange.bind(this);
    }

    handleClickOpen() {
        this.setState({ open: true });
    }

    handleClose() {
        this.setState({
            open: false,
            noteTitle: '',
            noteDescription: '',
        });
        this.props.history.push('/home');
    }

    handleNoteTitleChange(event) {
        this.setState({ noteTitle: event.target.value });
    }

    handleNoteDescriptionChange(event) {
        this.setState({ noteDescription: event.target.value });
    }

    handleUpdateNote() {
        const updatedNote = {
            id: parseFloat(this.props.match.params.id),
            noteTitle: this.state.noteTitle,
            noteDescription: this.state.noteDescription,
            color: this.state.color,
        }
        this.props.handleUpdateNote(updatedNote);
        this.handleClose();
    }
    handleClick = () => {
        this.setState({ displayColorPicker: !this.state.displayColorPicker })
    };

    handleColorClose = () => {
        this.setState({ displayColorPicker: false })
    };

    handleChange = (color) => {
        this.setState({ color: color.rgb, background: color.hex })
    };
    render() {
        const { classes } = this.props;
        const cover = {
            position: 'fixed',
            top: '0px',
            right: '0px',
            bottom: '0px',
            left: '0px',
        }
        return (
            <Dialog
                open={this.state.open}
                onClose={this.handleClose}
                aria-labelledby="Edit note form"
                style={{background: `rgba(${ this.state.color.r }, ${ this.state.color.g }, ${ this.state.color.b }, ${ this.state.color.a })`}} 
            >
                <DialogTitle id="Edit note form">
                    <div style={{ display: 'flex', justifyContent: 'space-between' }}>
                        Edit Note
                            <div style={styles.swatch} onClick={this.handleClick}>
                            <ColorLensIcon className={classes.iconHover} style={{ fontSize: 30, color: this.state.background }} />
                        </div>
                        {this.state.displayColorPicker ? <div className={classes.popover}>
                            <div style={cover} onClick={this.handleColorClose} />
                            <ColorPicker color={this.state.color}
                                onChangeComplete={this.handleChange} />
                        </div> : null}
                    </div>
                </DialogTitle>
                <DialogContent>
                    <TextField
                        autoFocus
                        margin="dense"
                        id="note title"
                        label="Note Title"
                        type="text"
                        fullWidth
                        onChange={this.handleNoteTitleChange}
                        value={this.state.noteTitle}
                    />
                    <TextField
                        margin="dense"
                        id="note description"
                        label="Note Description"
                        type="text"
                        onChange={this.handleNoteDescriptionChange}
                        value={this.state.noteDescription}
                        fullWidth
                    />
                </DialogContent>
                <DialogActions>
                    <Button onClick={this.handleClose} color="primary">
                        Cancel
                        </Button>
                    <Button onClick={this.handleUpdateNote} color="primary">
                        Update
                    </Button>
                </DialogActions>
            </Dialog>
        );
    }
}

export default withStyles(styles)(EditNote);