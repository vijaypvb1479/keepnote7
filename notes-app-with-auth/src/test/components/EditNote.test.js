import React from 'react';
import {mount } from 'enzyme';
import EditNote from '../../components/EditNote';

import  {createSerializer} from 'enzyme-to-json';
 
expect.addSnapshotSerializer(createSerializer({mode: 'deep'}));

let notes, handleRemoveNote, wrapper;

beforeEach(() => {
    const match = { params: { id: '1' } }
    notes = [
        {
            "noteTitle": "note-1",
            "noteDescription": "note-1 description",
            "id": 2
        }, {
            "noteTitle": "note-2",
            "noteDescription": "note-2 description",
            "id": 1
        }];

    wrapper = mount(
        <EditNote
            notes={notes}
            match={match} />
    );

})

test('test Grid View ', () => {

    expect(wrapper).toMatchSnapshot();

});