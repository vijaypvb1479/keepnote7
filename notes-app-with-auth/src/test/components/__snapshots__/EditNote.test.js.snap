// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Grid View  1`] = `
<Portal
  containerInfo={
    <body
      style="overflow: hidden; padding-right: 0px;"
    >
      <div
        aria-labelledby="Edit note form"
        class="mui-fixed MuiModal-root-15 MuiDialog-root-1"
        role="dialog"
        style="padding-right: 0px;"
      >
        <div
          aria-hidden="true"
          class="MuiBackdrop-root-17"
          style="opacity: 1; -webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
        />
        <div
          class="MuiDialog-container-4 MuiDialog-scrollPaper-2"
          role="document"
          style="opacity: 1; -webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
          tabindex="-1"
        >
          <div
            class="MuiPaper-root-19 MuiPaper-elevation24-45 MuiPaper-rounded-20 MuiDialog-paper-5 MuiDialog-paperScrollPaper-6 MuiDialog-paperWidthSm-9"
          >
            <div
              class="MuiDialogTitle-root-46"
              id="Edit note form"
            >
              <h2
                class="MuiTypography-root-47 MuiTypography-title-53"
              >
                Edit Note
              </h2>
            </div>
            <div
              class="MuiDialogContent-root-82"
            >
              <div
                class="MuiFormControl-root-83 MuiFormControl-marginDense-85 MuiFormControl-fullWidth-86"
              >
                <label
                  class="MuiFormLabel-root-98 MuiFormLabel-filled-102 MuiFormLabel-focused-99 MuiInputLabel-focused-88 MuiInputLabel-root-87 MuiInputLabel-formControl-92 MuiInputLabel-animated-95 MuiInputLabel-shrink-94 MuiInputLabel-marginDense-93"
                  data-shrink="true"
                  for="note title"
                >
                  Note Title
                </label>
                <div
                  class="MuiInputBase-root-118 MuiInput-root-105 MuiInput-underline-109 MuiInputBase-fullWidth-127 MuiInput-fullWidth-112 MuiInputBase-focused-120 MuiInput-focused-107 MuiInputBase-formControl-119 MuiInput-formControl-106 MuiInputBase-marginDense-125"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input-128 MuiInput-input-113 MuiInputBase-inputMarginDense-129 MuiInput-inputMarginDense-114"
                    id="note title"
                    type="text"
                    value="note-2"
                  />
                </div>
              </div>
              <div
                class="MuiFormControl-root-83 MuiFormControl-marginDense-85 MuiFormControl-fullWidth-86"
              >
                <label
                  class="MuiFormLabel-root-98 MuiFormLabel-filled-102 MuiInputLabel-root-87 MuiInputLabel-formControl-92 MuiInputLabel-animated-95 MuiInputLabel-shrink-94 MuiInputLabel-marginDense-93"
                  data-shrink="true"
                  for="note description"
                >
                  Note Description
                </label>
                <div
                  class="MuiInputBase-root-118 MuiInput-root-105 MuiInput-underline-109 MuiInputBase-fullWidth-127 MuiInput-fullWidth-112 MuiInputBase-formControl-119 MuiInput-formControl-106 MuiInputBase-marginDense-125"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input-128 MuiInput-input-113 MuiInputBase-inputMarginDense-129 MuiInput-inputMarginDense-114"
                    id="note description"
                    type="text"
                    value="note-2 description"
                  />
                </div>
              </div>
            </div>
            <div
              class="MuiDialogActions-root-135"
            >
              <button
                class="MuiButtonBase-root-163 MuiButton-root-137 MuiButton-text-139 MuiButton-textPrimary-140 MuiButton-flat-142 MuiButton-flatPrimary-143 MuiDialogActions-action-136"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label-138"
                >
                  Cancel
                </span>
                <span
                  class="MuiTouchRipple-root-166"
                />
              </button>
              <button
                class="MuiButtonBase-root-163 MuiButton-root-137 MuiButton-text-139 MuiButton-textPrimary-140 MuiButton-flat-142 MuiButton-flatPrimary-143 MuiDialogActions-action-136"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label-138"
                >
                  Update
                </span>
                <span
                  class="MuiTouchRipple-root-166"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </body>
  }
>
  <div
    aria-labelledby="Edit note form"
    className="mui-fixed MuiModal-root-15 MuiDialog-root-1"
    role="dialog"
  >
    <div
      aria-hidden="true"
      className="MuiBackdrop-root-17"
      onClick={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    />
    <div
      className="MuiDialog-container-4 MuiDialog-scrollPaper-2"
      onClick={[Function]}
      role="document"
      style={
        Object {
          "opacity": 1,
        }
      }
      tabIndex="-1"
    >
      <div
        className="MuiPaper-root-19 MuiPaper-elevation24-45 MuiPaper-rounded-20 MuiDialog-paper-5 MuiDialog-paperScrollPaper-6 MuiDialog-paperWidthSm-9"
      >
        <div
          className="MuiDialogTitle-root-46"
          id="Edit note form"
        >
          <h2
            className="MuiTypography-root-47 MuiTypography-title-53"
          >
            Edit Note
          </h2>
        </div>
        <div
          className="MuiDialogContent-root-82"
        >
          <div
            className="MuiFormControl-root-83 MuiFormControl-marginDense-85 MuiFormControl-fullWidth-86"
          >
            <label
              className="MuiFormLabel-root-98 MuiFormLabel-filled-102 MuiFormLabel-focused-99 MuiInputLabel-focused-88 MuiInputLabel-root-87 MuiInputLabel-formControl-92 MuiInputLabel-animated-95 MuiInputLabel-shrink-94 MuiInputLabel-marginDense-93"
              data-shrink={true}
              htmlFor="note title"
            >
              Note Title
            </label>
            <div
              className="MuiInputBase-root-118 MuiInput-root-105 MuiInput-underline-109 MuiInputBase-fullWidth-127 MuiInput-fullWidth-112 MuiInputBase-focused-120 MuiInput-focused-107 MuiInputBase-formControl-119 MuiInput-formControl-106 MuiInputBase-marginDense-125"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={true}
                className="MuiInputBase-input-128 MuiInput-input-113 MuiInputBase-inputMarginDense-129 MuiInput-inputMarginDense-114"
                disabled={false}
                id="note title"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
                value="note-2"
              />
            </div>
          </div>
          <div
            className="MuiFormControl-root-83 MuiFormControl-marginDense-85 MuiFormControl-fullWidth-86"
          >
            <label
              className="MuiFormLabel-root-98 MuiFormLabel-filled-102 MuiInputLabel-root-87 MuiInputLabel-formControl-92 MuiInputLabel-animated-95 MuiInputLabel-shrink-94 MuiInputLabel-marginDense-93"
              data-shrink={true}
              htmlFor="note description"
            >
              Note Description
            </label>
            <div
              className="MuiInputBase-root-118 MuiInput-root-105 MuiInput-underline-109 MuiInputBase-fullWidth-127 MuiInput-fullWidth-112 MuiInputBase-formControl-119 MuiInput-formControl-106 MuiInputBase-marginDense-125"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                className="MuiInputBase-input-128 MuiInput-input-113 MuiInputBase-inputMarginDense-129 MuiInput-inputMarginDense-114"
                disabled={false}
                id="note description"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
                value="note-2 description"
              />
            </div>
          </div>
        </div>
        <div
          className="MuiDialogActions-root-135"
        >
          <button
            className="MuiButtonBase-root-163 MuiButton-root-137 MuiButton-text-139 MuiButton-textPrimary-140 MuiButton-flat-142 MuiButton-flatPrimary-143 MuiDialogActions-action-136"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex="0"
            type="button"
          >
            <span
              className="MuiButton-label-138"
            >
              Cancel
            </span>
            <span
              className="MuiTouchRipple-root-166"
            />
          </button>
          <button
            className="MuiButtonBase-root-163 MuiButton-root-137 MuiButton-text-139 MuiButton-textPrimary-140 MuiButton-flat-142 MuiButton-flatPrimary-143 MuiDialogActions-action-136"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex="0"
            type="button"
          >
            <span
              className="MuiButton-label-138"
            >
              Update
            </span>
            <span
              className="MuiTouchRipple-root-166"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</Portal>
`;
